### PR TITLE
Factorized various eval_as_xxx into a single eval_as template method

### PR DESCRIPTION
**Pull request by biagiop1986**
I factorized three different similar eval_as_xxx methods into a single template eval_as.

### CODE REVIEW

**Comment by giubacc** [comment]
Hello @biagiop1986! Many thanks for your contribution!
The patch you are proposing is of the highest quality! :top: :top: :top:

Let's try to get rid of the things that must be corrected:

- `commit signature`: you are committing without a proper signature. Please, take a look [here](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification). The best option is that you generate a new GPG key and you upload the public key on a public key server.
- `Lint` check is failing. You can check details by clicking on the relevant CI item. The problem is that there are some trailing whitespaces, you should format source files with [astyle](https://astyle.sourceforge.net/). Most distributions provide directly the package. Once you have installed it, you should use the [config](https://github.com/giubacc/chatterbox/blob/6c680462990c2cfa03e1e9b04a92b8e389024032/src/.astylerc) file under src and execute: 
```
cd ${workspaceFolder}/src && astyle --project=.astylerc *.cpp *.h
```
- `Build` CI item is failing. This happens because currently the CI is not properly configured to handle correctly contributions. The problem is that  [build-builder.yaml](https://github.com/giubacc/chatterbox/blob/6c680462990c2cfa03e1e9b04a92b8e389024032/.github/workflows/build-builder.yaml) is currently configured to run on Sunday night. Because you opened this PR before this event ever run on your forked repository, there is no such image that CI can use to build `chatterbox`. A possible workaround to this could be that you try to build the `builder` image on your machine and then you push it on github. Let's take this issue for last.

**Comment by giubacc** [comment]
@biagiop1986 you need to sync your fork and then rebase against main.
Now you have the `chatterbox-builder` image available, so CI should not be broken anymore.

**Comment by biagiop1986** [comment]
> @biagiop1986 you need to sync your fork and then rebase against main. Now you have the `chatterbox-builder` image available, so CI should not be broken anymore.

Thank you, done right now! I'm going to open a new pull request.

File: src/chatterbox.h [hunk 1]
```diff
@@ -225,4 +286,62 @@ struct chatterbox {
...
+  }
+};
+
+template <>
+struct chatterbox::Converter<std::string> {
+  static bool isType(const Json::Value &val) {
+    return val.isString();
+  }
+  static bool isType(const v8::Local<v8::Value> &val) {
+    return val->IsUint32();
```
Comment (giubacc) [inline src/chatterbox.h:333]:
```suggestion
    return val->IsString();
```

**Review by giubacc — CHANGES_REQUESTED** [review]
just a small fix; otherwise LGTM

File: src/chatterbox.h (hunk 1 above)
Comment (biagiop1986) [reply src/chatterbox.h]: Fixed!

**Comment by giubacc** [comment]
please, squash the 2 commits, no need to co-author the latest commit.

**Comment by biagiop1986** [comment]
> please, squash the 2 commits, no need to co-author the latest commit.

Done!

**Comment by giubacc** [comment]
thanks! merged :100:

**Pull request merged**